blog/_posts/2021-01-11-scala-native-0.4-release.md

```diff
@@ -0,0 +1,273 @@
+you can use ad-hoc sbt setting:
+```bash
+sbt 'set nativeConfig ~= {_.withTargetTriple("x86_64-apple-darwin<version>")}' myApp/nativeLink
+```
+
+We consider changing target triple as a feature for advanced users, and cannot promise it would currently work with any possible configuration yet.
+However, the number of supported architectures and operating systems would definitely grow in the future.
+
+> Upcoming release will no longer support SBT 0.13.x
+
+## Changes to intermediate representation
+The new release includes many necessary changes to NIR, starting with the way Strings are stored, 
+through adding new literals like `Val.ClassOf` and last, but not least support for default methods. 
+The last one was introduced to support Scala 2.12 behavior which compiles trait methods as default methods in Java interfaces.
+
+> NIR (Native Intermediate Representation) is a format used to represent a subset of LLVM instructions, types and values, 
+> augmented with a number of high-level primitives that are necessary to efficiently compile Scala,
+> you can read more about in Scala Native [documentation](https://scala-native.readthedocs.io/en/latest/contrib/nir.html).
+
+To keep the NIR changes to a minimum, we continue generating a `FunctionN` class with an `apply` method for each lambda. 
+All the lambda captures are stored inside the class instance and initialized via its constructor. 
+The `apply` method calls the generated `$anonfun` method, also extracting and forwarding the captures.
+Scala Native does not have any special treatment for closures. It uses the reimplemented anonymous class generation which
+was originally used in earlier versions. Each anonymous functions gets a generated class similarly to closure encoding 
+on Scala 2.11 and earlier
+
+```scala
+class AnonFunX extends java.lang.Object with FunctionalInterface
+  var capture1: T1
+  ...
+  var captureN: TN
+
+  def <init>(this, v1: T1, ..., vN: TN): Unit = {
+    java.lang.Object.<init>(this)
+    this.capture1 = v1
+     ...
+    this.captureN = vN
+  }
+
+  def samMethod(param1: Tp1, ..., paramK: TpK): Tret = {
+    EnclosingClass.this.staticMethod(
+       param1, ..., paramK,
+       this.capture1, ..., this.captureN
+    )
+  }
+}
+```
+
+
+## How about existing libraries?
+Scala Native had many [libraries](https://index.scala-lang.org/search?targetTypes=native&scalaNativeVersions=0.4.0-M2) compatible with SN `0.4.0-M2`.
+Unfortunately, the new release is not binary compatible with them, and you'll need to wait until authors will republish them with the new SN version.
+
+However, with increasing interest in running the Scala code natively, it should happen quickly after the release.
+
+## Summary
+Even though Scala Native was covered in dust in the past, it starts to keep up with the rest of the Scala ecosystem. 
```

Comment:
I feel like the blog is a bit too negative about the prior status of SN. I think it’s ok to acknowledge that there were no resources allocated to it until recently, and that we now have 1 engineer working full time on it. I think it would be good to also link to https://contributors.scala-lang.org/t/scala-native-next-steps/4216 somewhere.